### PR TITLE
Dashboard API improvements(#1392)

### PR DESCRIPTION
**Pull request by juhijariwala**

### CODE REVIEW

**Comment by ketan** [comment]
There are some more changes coming. Please hold off merging this.

**Comment by zabil** [comment]
@ketan @juhi this is good to go right?

File: server/webapp/WEB-INF/rails.new/spec/helpers/application_helper_spec.rb [hunk 1]
```diff
@@ -40,29 +40,29 @@
 
     it "should handle default_url_options" do
       url = url_for_path("/foo")
-      url.should == "/go/quux/foo?x"
+      url.should == "/foo"
```
Comment (ketan) [inline server/webapp/WEB-INF/rails.new/spec/helpers/application_helper_spec.rb]: None of these tests should have changed.

**Comment by ketan** [comment]
@juhijariwala - can you also submit additional PRs to -

* remove old "undocumented" dashboard API, associated controllers, views, helpers and any java code. Looking at @arvindsv's code (#373 and #371( that should tell you what you'd need to revert `git log --grep '#37[13]\b' --name-only` and `git log --grep '#37[13]\b' -p` should tell you most relevant code :)
* documentation over at api.go.cd

In addition, please make sure that the documentation links in the current PR are correct (or ensure that the future API docs will be available at the URL)

**Pull request merged**